Source: Written by Ali Polatel <alip@exherbo.org>
Upstream: Not yet
Reason: Make the ltree extension build optional via --without-ltree
Comment: Originally written for 9.6.3, updated for 11.3
diff -ur postgresql-11.3/configure.in postgresql-11.3-ltree/configure.in
--- postgresql-11.3/configure.in	2019-05-06 22:46:18.000000000 +0200
+++ postgresql-11.3-ltree/configure.in	2019-05-09 16:56:00.483340618 +0200
@@ -869,6 +869,14 @@
 PGAC_ARG_BOOL(with, libedit-preferred, no,
               [prefer BSD Libedit over GNU Readline])
 
+#
+# Optionally build the ltree extension (default: yes)
+#
+AC_MSG_CHECKING([whether to build the ltree extension])
+PGAC_ARG_BOOL(with, ltree, yes, [build the ltree extension])
+AC_MSG_RESULT([$with_ltree])
+AC_SUBST(with_ltree)
+
 
 #
 # UUID library
diff -ur postgresql-11.3/contrib/Makefile postgresql-11.3-ltree/contrib/Makefile
--- postgresql-11.3/contrib/Makefile	2019-05-06 22:46:18.000000000 +0200
+++ postgresql-11.3-ltree/contrib/Makefile	2019-05-09 16:57:06.114143916 +0200
@@ -25,7 +25,6 @@
 		intarray	\
 		isn		\
 		lo		\
-		ltree		\
 		oid2name	\
 		pageinspect	\
 		passwordcheck	\
@@ -50,6 +49,12 @@
 		unaccent	\
 		vacuumlo
 
+ifeq ($(with_ltree),yes)
+SUBDIRS += ltree
+else
+ALWAYS_SUBDIRS += ltree
+endif
+
 ifeq ($(with_openssl),yes)
 SUBDIRS += sslinfo
 else
@@ -81,7 +86,10 @@
 endif
 
 ifeq ($(with_python),yes)
-SUBDIRS += hstore_plpython jsonb_plpython ltree_plpython
+SUBDIRS += hstore_plpython jsonb_plpython
+ifeq ($(with_ltree),yes)
+SUBDIRS += ltree_plpython
+endif
 else
 ALWAYS_SUBDIRS += hstore_plpython jsonb_plpython ltree_plpython
 endif
diff -ur postgresql-11.3/src/Makefile.global.in postgresql-11.3-ltree/src/Makefile.global.in
--- postgresql-11.3/src/Makefile.global.in	2019-05-06 22:46:18.000000000 +0200
+++ postgresql-11.3-ltree/src/Makefile.global.in	2019-05-09 16:57:34.194488765 +0200
@@ -193,6 +193,7 @@
 with_libxml	= @with_libxml@
 with_libxslt	= @with_libxslt@
 with_llvm	= @with_llvm@
+with_ltree	= @with_ltree@
 with_system_tzdata = @with_system_tzdata@
 with_uuid	= @with_uuid@
 with_zlib	= @with_zlib@
